lib/API/MP/v1/API.php

```diff
@@ -217,17 +232,9 @@ function addSubscriber(array $subscriber, $segments = array(), $options = array(
 
     // subscribe to segments and optionally: 1) send confirmation email, 2) schedule welcome email(s)
     if (!empty($segments)) {
-      $this->subscribeToLists($new_subscriber->id, $segments);
-
-      // send confirmation email
-      if ($send_confirmation_email && $new_subscriber->status === Subscriber::STATUS_UNCONFIRMED) {
-        $result = $this->_sendConfirmationEmail($new_subscriber);
-        if (!$result && $new_subscriber->getErrors()) {
-          throw new \Exception(
-            WPFunctions::get()->__(sprintf('Subscriber added, but confirmation email failed to send: %s', strtolower(implode(', ', $new_subscriber->getErrors()))), 'mailpoet')
-          );
-        }
-      }
+      $this->subscribeToLists($new_subscriber->id, $segments, [
```

Comment:
Here we fetch the subscriber from the DB then we pass his id to `subscribeToLists` which fetches him again from DB. Is there a way to not do the same DB query twice?

Reply:
Good point. I'm afraid there is no easy way to fix that now. Considering:
1. It was like this already before, I didn't change that behavior (I didn't add `subscribeToLists`, only added params) .
2. Some ORM's would cache the subscriber per request (Doctrine).

Do you think it needs a fix? I cannot think of a one.

Reply:
A fix is possible but it requires changing the API interface, and since we will switch to a better ORM with caching feature in the future I think this is not a big deal.